src/solana/index.ts

```diff
@@ -340,50 +376,400 @@ export class Solana extends Blockchain {
+      stakePubkey: stakeAccount.publicKey,
+      lockup: lockup,
+    });
+    createStakeAccountTx.recentBlockhash = blockhash;
+    createStakeAccountTx.sign(stakeAccount);
+
+    return [createStakeAccountTx, stakeAccount.publicKey, [stakeAccount]];
+  }
+
+  private async createAccountWithSeedTx(
+    authorityPublicKey: PublicKey,
+    lamports: number,
+    source: string,
+    lockup: Lockup,
+  ): Promise<[Transaction, PublicKey, Keypair[]]> {
+    // Format source to
+    const seed = this.formatSource(source);
+
+    const stakeAccountPubkey = await PublicKey.createWithSeed(
+      authorityPublicKey,
+      seed,
+      StakeProgram.programId,
+    );
+
+    const createStakeAccountTx = new Transaction().add(
+      StakeProgram.createAccountWithSeed({
+        authorized: new Authorized(authorityPublicKey, authorityPublicKey),
+        fromPubkey: authorityPublicKey,
+        basePubkey: authorityPublicKey,
+        stakePubkey: stakeAccountPubkey,
+        lockup: lockup,
+
+        seed: seed,
+        lamports: lamports,
+      }),
+    );
+
+    return [createStakeAccountTx, stakeAccountPubkey, []];
+  }
+
+  /** unstake - unstake
+   * @param {string} token - auth API token
+   * @param {string} sender - account blockchain address (staker)
+   * @param {number} lamports - lamport amount
+   * @param {string} source - stake source
+   * @returns {Promise<object>} Promise object with Versioned Tx
+   */
+  public async unstake(
+    token: string,
+    sender: string,
+    lamports: number,
+    source: string,
+  ): Promise<ApiResponse<VersionedTransaction>> {
+    // Check if the token is valid
+    const isTokenValid = await CheckToken(token);
+    if (!isTokenValid) {
+      this.throwError('INVALID_TOKEN_ERROR');
+    }
+
+    try {
+      const delegations = await this.getDelegations(sender);
+
+      const stakeAccounts = delegations.result.map((delegationAcc) => {
+        return {
+          pubkey: delegationAcc.pubkey,
+          account: parsedAccountInfoToStakeAccount(delegationAcc.account),
+        };
+      });
+
+      const epochInfo = await this.connection.getEpochInfo();
+      const tm = this.timestampInSec();
+
+      let totalActiveStake = new BigNumber(0);
+      const activeStakeAccounts = stakeAccounts.filter((acc) => {
+        const isActive = !(
+          isLockupInForce(acc.account.data.info.meta, epochInfo.epoch, tm) ||
+          stakeAccountState(acc.account.data, epochInfo.epoch) !==
+            StakeState.Active
+        );
+        if (isActive && acc.account.data.info.stake !== null) {
+          totalActiveStake = totalActiveStake.plus(
+            acc.account.data.info.stake.delegation.stake,
+          );
+        }
+
+        return isActive;
+      });
+
+      let lamportsBN = new BigNumber(lamports);
+      if (totalActiveStake.lt(lamportsBN))
+        throw this.throwError('NOT_ENOUGH_ACTIVE_STAKE_ERROR');
+
+      // Desc sorting
+      activeStakeAccounts.sort((a, b): number => {
```

Comment:
```suggestion
  activeStakeAccounts.sort((a, b) => {
    const stakeA = a.account.data.info.stake?.delegation.stake;
    const stakeB = b.account.data.info.stake?.delegation.stake;

    if (!stakeA || !stakeB) return 0;

    return stakeB.minus(stakeA).toNumber();
});
```
I think this would also work. Not a blocker.